File: 백준/Bronze/9933. 민균이의 비밀번호/민균이의 비밀번호.py
import sys as s
n = int(s.stdin.readline().strip())
arr = []
reverse_arr = []

for i in range(n):
    arr.append(str(s.stdin.readline().strip()))
    reverse_arr.append(arr[i][::-1])

for i in arr:
    for j in reverse_arr:
        if i == j:
            a = i
            length = len(a)
            middle_char = a[length//2]
            print(length, middle_char)
            exit()

File: 백준/Bronze/9933. 민균이의 비밀번호/민균이의 비밀번호.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        
        ArrayList<String> arr = new ArrayList<>();
        ArrayList<String> reverseArr = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            String str = br.readLine().trim();
            arr.add(str);
            reverseArr.add(new StringBuilder(str).reverse().toString());
        }

        for (String original : arr) {
            for (String reversed : reverseArr) {
                if (original.equals(reversed)) {
                    int length = original.length();
                    char middleChar = original.charAt(length / 2);
                    System.out.println(length + " " + middleChar);
                    return;
                }
            }
        }
    }
}